Sources/Swift/Helper/SentryCurrentDateProvider.swift

```diff
@@ -1,7 +1,28 @@
 @_implementationOnly import _SentryPrivate
 import Foundation
 
-@objcMembers class SentryCurrentDateProvider: NSObject {
+#if TEST
```

Comment:
It's an ugly solution for our unusual project setup.

Im also fine im making this class public, I dont think it would be a problem.

Reply:
`m`: Wouldn't defining a protocol also work? The advantage would be that it's easier to notice when the two get out of sync, cause it will throw a compiler error. Also, it's easy add different implementations of `SentryCurrentDateProvider` for the tests.

```swift
@objc protocol SentryCurrentDateProvider {
    func date() -> Date
    func timezoneOffset() -> Int
    func systemTime() -> UInt64
    func systemUptime() -> TimeInterval
}

@objcMembers
class SentryDefaultCurrentDateProvider: NSObject, SentryCurrentDateProvider {
    
    func date() -> Date {
        return Date()
    }

    func timezoneOffset() -> Int {
        return TimeZone.current.secondsFromGMT()
    }

    func systemTime() -> UInt64 {
        getAbsoluteTime()
    }
    
    func systemUptime() -> TimeInterval {
        ProcessInfo.processInfo.systemUptime
    }
}

Reply:
It seems that the same problem does not occur for a private Protocol
Good!!

Its a bigger refactoring tho.